crates/iceberg-catalog/src/implementations/postgres/tabular/table.rs

```diff
@@ -334,79 +337,84 @@ pub(crate) async fn get_table_metadata_by_id(
     })
 }
 
-pub(crate) async fn get_table_metadata_by_s3_location(
+pub(crate) async fn get_table_id_by_s3_location(
     warehouse_id: WarehouseIdent,
     location: &str,
```

Comment:
Could we change the type to configs::Location (basically URL?).
I think just by splitting location at :// it is not guaranteed to be a valid URL. Location type is a bit more typesafe and allows easier access to protocol and (sub)-paths by `path_segemnts`.
We might need to expose more attributes of Location.
Maybe having sublocation part of configs.Location makes sense? Then we could share it across impls again.

Reply:
changed to use location instead